### PR TITLE
RUN-4372 IAB handler to TypeScript

### PR DESCRIPTION
**Pull request by whyn07m3**
ℹ️ **About**
This PR transitions IAB API handler from JavaScript to TypeScript

⚠️ Yes, used `git mv` to rename the file.

🚥 **Automated test results**
✅ [Windows 7](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5b71bd0a54b21953031f395e) (same fails as [vanilla version](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5b71bbc854b21953031f395c) )
✅ [Windows 10](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5b71be1154b21953031f395f) (same fails as [vanilla version](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5b71bcf154b21953031f395d) )

### CODE REVIEW

File: src/browser/api_protocol/api_handlers/interappbus.ts [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+import ofEvents from '../../of_events';
+import route from '../../../common/route';
+import {
+    Acker,
+    APIMessage,
+    Identity,
+    Subscriber
+} from '../../../shapes';
+
+const subScriptionTypes = {
```
Comment (pbaize) [inline src/browser/api_protocol/api_handlers/interappbus.ts]: This should probably be a `const enum`

File: src/browser/api_protocol/api_handlers/interappbus.ts [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+
+        // old subscribing to new
+        if (!subscribedMessageKey && (sentMessageKey === 'messageString')) {
+            command.payload.message = JSON.parse(payload[sentMessageKey]);
+        }
+
+        apiProtocolBase.sendToIdentity(identity, command);
+    };
+
+    const subscriptionArgs = [
```
Comment (pbaize) [inline src/browser/api_protocol/api_handlers/interappbus.ts]: Should we define a tuple type for IAB subscription args? They always have the same shape and this is the only type `any` in this file

File: src/browser/api_protocol/api_handlers/interappbus.ts [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+import {
+    Acker,
+    APIMessage,
+    Identity,
+    Subscriber
+} from '../../../shapes';
+
+type IABSubscriptionArgs = [string, string, string, string, number];
+
+enum SubScriptionTypes {
```
Comment (HarsimranSingh) [inline src/browser/api_protocol/api_handlers/interappbus.ts]: :neckbeard: subscriptionTypes

Comment (whyn07m3) [reply src/browser/api_protocol/api_handlers/interappbus.ts]: Changed it to `SubscriptionTypes` though

**Pull request merged**